### PR TITLE
machine: silently cleanup dangling sockets before `rm` if possible

### PR DESCRIPTION
**Pull request by flouthoc**
Try to cleanup dandling pid and machine socket if possible silently
before `rm`.

Closes: https://github.com/containers/podman/issues/11393

[NO TESTS NEEDED]

### CODE REVIEW

**Comment by flouthoc** [comment]
@Luap99 PTAL

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -460,6 +463,20 @@ func (v *MachineVM) Remove(name string, opts machine.RemoveOptions) (string, fun
 	for _, msg := range files {
 		confirmationMessage += msg + "\n"
 	}
+
+	//silently try to delete socket and pid file
+	//remove socket and pid file if any: warn at low priority if things fail
+	if errSocketFile != nil {
```
Comment (mheon) [inline pkg/machine/qemu/machine.go:469]: Why `!= nil`?

File: pkg/machine/qemu/machine.go [hunk 2]
```diff
@@ -432,6 +432,9 @@ func (v *MachineVM) Remove(name string, opts machine.RemoveOptions) (string, fun
 		return "", nil, errors.Errorf("running vm %q cannot be destroyed", v.Name)
 	}
 
+	// Get path to socket and pidFile before we do any cleanups
+	qemuSocketFile, pidFile, errSocketFile := v.getSocketandPid()
+
```
Comment (Luap99) [inline pkg/machine/qemu/machine.go:437]: Why do you call it here and only use it 30 lines later, please move this down.

Comment (flouthoc) [reply pkg/machine/qemu/machine.go]: @Luap99 I thought rumtime would be removed and become invalid after `RemoveConnection` but just checked everything is static so cool :) Fixed.

**Comment by Luap99** [comment]
@baude PTAL

File: pkg/machine/qemu/machine.go [hunk 3]
```diff
@@ -460,6 +460,22 @@ func (v *MachineVM) Remove(name string, opts machine.RemoveOptions) (string, fun
...
+	qemuSocketFile, pidFile, errSocketFile := v.getSocketandPid()
+	//silently try to delete socket and pid file
+	//remove socket and pid file if any: warn at low priority if things fail
+	if errSocketFile == nil {
+		// Remove the pidfile
+		if err := os.Remove(pidFile); err != nil && !errors.Is(err, os.ErrNotExist) {
+			logrus.Debugf("Error while removing pidfile: %v", err)
+		}
+		// Remove socket
+		if err := os.Remove(qemuSocketFile); err != nil {
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:474]: Why not check ErrNotExist here?

Comment (flouthoc) [reply pkg/machine/qemu/machine.go]: yes we must check for ErrNotExist here as well , thanks

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests and I'd like a @baude head nod/merge

**Comment by flouthoc** [comment]
@baude @containers/podman-maintainers PTAL

**Pull request merged**